application/app/src/test/java/uk/nhs/digital/uec/api/integration/fuzzysearch/FuzzySearchSuccessTest.java

```diff
@@ -0,0 +1,163 @@
+
+    ApiSuccResponse apiSuccResponse = mapper.readValue(response.getBody(), ApiSuccResponse.class);
+
+    assertEquals(apiSuccResponse.getNumberOfServices(), 2);
+
+    // Add the service ids that you want to verify
+    List<Integer> serviceIdsToVerify = new ArrayList<>();
+    serviceIdsToVerify.add(1);
+    serviceIdsToVerify.add(2);
+
+    verifySuccessResponse(apiSuccResponse, serviceIdsToVerify);
+  }
+
+  /**
+   * Given at least one valid search term, ensure the API returns a 200 success code, and a list of
+   * services are returned.
+   *
+   * @throws Exception
+   */
+  @Test
+  public void atLeastOneValidSearchTermGiven() throws Exception {
+    HttpEntity<String> request = new HttpEntity<String>(null, headers);
+    UriComponentsBuilder uriBuilder =
+        UriComponentsBuilder.fromHttpUrl(endpointUrl)
+            .queryParam("search_criteria", "a")
+            .queryParam("search_criteria", "ab")
+            .queryParam("search_criteria", "Term1");
+    ResponseEntity<String> response =
+        restTemplate.exchange(uriBuilder.toUriString(), HttpMethod.GET, request, String.class);
+
+    assertTrue(response.getStatusCode() == HttpStatus.OK);
+
+    ApiSuccResponse apiSuccResponse = mapper.readValue(response.getBody(), ApiSuccResponse.class);
+
+    assertEquals(apiSuccResponse.getNumberOfServices(), 2);
+
+    // Add the service ids that you want to verify
+    List<Integer> serviceIdsToVerify = new ArrayList<>();
+    serviceIdsToVerify.add(1);
+    serviceIdsToVerify.add(2);
+
+    verifySuccessResponse(apiSuccResponse, serviceIdsToVerify);
+  }
+
+  /**
+   * Given valid search criteria terms, ensure the API returns a 200 success code, and a list of no
+   * services are returned.
+   *
+   * @throws Exception
+   */
+  @Test
+  public void oneSearchCriteriaGivenNoResults() throws Exception {
+    HttpEntity<String> request = new HttpEntity<String>(null, headers);
+    UriComponentsBuilder uriBuilder =
+        UriComponentsBuilder.fromHttpUrl(endpointUrl).queryParam("search_criteria", "Term0");
+    ResponseEntity<String> response =
+        restTemplate.exchange(uriBuilder.toUriString(), HttpMethod.GET, request, String.class);
+
+    assertTrue(response.getStatusCode() == HttpStatus.OK);
+
+    ApiSuccResponse apiSuccResponse = mapper.readValue(response.getBody(), ApiSuccResponse.class);
+
+    assertEquals(apiSuccResponse.getNumberOfServices(), 0);
+  }
+
+  /** Method to verify the service details that are returned from the API call. */
+  private void verifySuccessResponse(
+      ApiSuccResponse apiSuccResponse, List<Integer> serviceIdsToVerify) {
+
+    for (Integer serviceIdToVerify : serviceIdsToVerify) {
+
+      DosService dosServiceToCheckAgainst =
+          MockDosServicesUtil.mockDosServices.get(serviceIdToVerify);
+
+      for (DosService dosServiceToVerify : apiSuccResponse.getServices()) {
+        if (dosServiceToVerify.getId() == serviceIdToVerify) {
+          assertEquals(
```

Comment:
Can we move the whole comparison to an overloaded method in the `DosService` class i.e. `equals()`?

Reply:
I think you mean _override_ the equals() method.  We can, but we would need to override the hash() method too. However its prob a worthwhile thing to do.

Reply:
Yes, I meant override 😃  Nice one